src/components/FeatureViews/Availability/CreateOOOForm.js

```diff
@@ -1,152 +1,149 @@
-				},
-				"query": GET_UPCOMING_INSTRUCTOR_OOO,
-				"variables": { instructorID: AuthUser.user.id }
-			});
-
-		}
-	})
-
-	const SubmitNoticeChild = useRef();
-	const updateOOOFormState = (newState) => {
-		setOOOFormState(newState);
-	};
-
-	const handleNext = () => {
-		setActiveStep((prevActiveStep) => prevActiveStep + 1);
-	};
-
-	const handleBack = () => {
-		setActiveStep((prevActiveStep) => prevActiveStep - 1);
-	};
-
-	const handleClose = (event) => {
-		event.preventDefault()
-		setSubmitted(false)
-	}
-
-	const handleSubmit = () => {
-		createOOO({
-			variables: {
-				instructorId: AuthUser.user.id,
-				startDatetime: OOOFormState.startDate.toDate().toISOString(),
-				endDatetime: OOOFormState.endDate.toDate().toISOString(),
-				description: OOOFormState.description,
-			}
-		});
-	}
-
-
-
-	return (<OOOContext.Provider value={{ OOOFormState, updateOOOFormState }}>
-		<Grid container direction="row">
-			{/* <Grid item>
+    const [activeStep, setActiveStep] = useState(0);
+    const [OOOFormState, setOOOFormState] = useState(null);
+    const [submitted, setSubmitted] = useState(false);
+    const steps = getSteps();
+    const AuthUser = useSelector(({ auth }) => auth);
+    const [createOOO, createOOOResults] = useMutation(CREATE_INSTRUCTOR_OOO, {
+        onCompleted: () => {
+            setSubmitted(true);
+        },
+        update: (cache, { data }) => {
+            const newOOO = data.createInstructorOoo.instructorOoo;
+            const cachedOOO = cache.readQuery({
+                query: GET_UPCOMING_INSTRUCTOR_OOO,
+                variables: { instructorID: AuthUser.user.id },
+            }).instructorOoo;
+
+            const updatedCache = [...cachedOOO, newOOO];
+
+            cache.writeQuery({
+                data: {
+                    instructorOoo: updatedCache,
+                },
+                query: GET_UPCOMING_INSTRUCTOR_OOO,
+                variables: { instructorID: AuthUser.user.id },
+            });
+        },
+    });
+
+    const SubmitNoticeChild = useRef();
+    const updateOOOFormState = (newState) => setOOOFormState(newState);
+
+    const handleNext = () =>
+        setActiveStep((prevActiveStep) => prevActiveStep + 1);
+
+    const handleBack = () =>
+        setActiveStep((prevActiveStep) => prevActiveStep - 1);
+
+    const handleClose = (event) => {
+        event.preventDefault();
+        setSubmitted(false);
+    };
+
+    const handleSubmit = () => {
+        createOOO({
+            variables: {
+                instructorId: AuthUser.user.id,
+                startDatetime: OOOFormState.startDate.toDate().toISOString(),
+                endDatetime: OOOFormState.endDate.toDate().toISOString(),
+                description: OOOFormState.description,
+            },
+        });
+    };
+
+    return (
+        <OOOContext.Provider value={{ OOOFormState, updateOOOFormState }}>
+            <Grid container direction="row">
+                {/* <Grid item>
```

Comment:
We need to comment this out because the final design is a step feature. I believe we are waiting for a conflicts api to check if there will be schedule conflicts with the instructors OOO and their working hours.